src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -3480,5 +3486,425 @@ void F()
+            var lambda = exprs.SelectAsArray(e => GetLambdaSymbol(model, e)).Single();
+            var parameter = (SourceParameterSymbol)lambda.Parameters[0];
+            Assert.True(parameter.HasOptionalAttribute);
+            Assert.False(parameter.HasExplicitDefaultValue);
+            Assert.Equal(2, parameter.DefaultValueFromAttributes.Value);
+        }
+
+        [ConditionalFact(typeof(DesktopOnly))]
+        public void LambdaParameterAttributes_WellKnownAttributes()
+        {
+            var source =
+@"using System;
+using System.Runtime.CompilerServices;
+class Program
+{
+    static void Main()
+    {
+        Action<object> a1 = ([IDispatchConstant] object obj) => { };
+        Action<object> a2 = ([IUnknownConstant] object obj) => { };
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var exprs = tree.GetRoot().DescendantNodes().OfType<LambdaExpressionSyntax>().ToImmutableArray();
+            Assert.Equal(2, exprs.Length);
+            var lambdas = exprs.SelectAsArray(e => GetLambdaSymbol(model, e));
+            Assert.True(lambdas[0].Parameters[0].IsIDispatchConstant);
+            Assert.True(lambdas[1].Parameters[0].IsIUnknownConstant);
+        }
+
+        [Fact]
+        public void LambdaParameterAttributes_NullableAttributes()
+        {
+            var source =
+@"using System;
+using System.Diagnostics.CodeAnalysis;
+class Program
+{
+    static void Main()
+    {
+        Action<object> a1 = ([AllowNull][MaybeNullWhen(false)] object obj) => { };
+        Action<object, object> a2 = (object x, [NotNullIfNotNull(""x"")] object y) => { };
+    }
+}";
+            var comp = CreateCompilation(
+                new[] { source, AllowNullAttributeDefinition, MaybeNullWhenAttributeDefinition, NotNullIfNotNullAttributeDefinition },
+                parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var exprs = tree.GetRoot().DescendantNodes().OfType<LambdaExpressionSyntax>().ToImmutableArray();
+            Assert.Equal(2, exprs.Length);
+            var lambdas = exprs.SelectAsArray(e => GetLambdaSymbol(model, e));
+            Assert.Equal(FlowAnalysisAnnotations.AllowNull | FlowAnalysisAnnotations.MaybeNullWhenFalse, lambdas[0].Parameters[0].FlowAnalysisAnnotations);
+            Assert.Equal(new[] { "x" }, lambdas[1].Parameters[1].NotNullIfParameterNotNull);
+        }
+
+        private static LambdaSymbol GetLambdaSymbol(SemanticModel model, LambdaExpressionSyntax syntax)
+        {
+            return model.GetSymbolInfo(syntax).Symbol.GetSymbol<LambdaSymbol>();
+        }
```

Comment:
Some test ideas, if not covered already:
- Test semantic model on attributes (like LocalFunctionAttribute_Argument_SemanticModel)
- Enforcement of attribute targets (like LocalFunctionAttribute_BadAttributeLocation)
- Disallowed attributes (like LocalFunctionDisallowedAttributes and LocalFunctionDisallowedSecurityAttributes)
- Scoping rules on attributes

Reply:
Added tests adapted from the corresponding attribute tests in `LocalFunctionTests`. Thanks for the suggestions (and thanks @RikkiGibson for the local function tests).